Fixing module-level imports to work with pylint and JEDI-vim

phargogh opened the pull request:
This is a minor thing, but it's been a mild issue for _years_.  So it's fixed now and my autocomplete is working as expected in `vim` and PyLint is able to find the right attributes.

I'm really not sure how to test this, so the tests aren't updated.

RE:#266

phargogh committed:
Fixing module-level imports.

RE:#266

phargogh committed:
Improving an inline docstring. RE:#266

dcdenu4 reviewed (approved):
Thanks @phargogh , kind of annoying that we have to manually keep this list. I'm not sure why the tests are failing. I tested locally with Python 3.9, gdal 3.4.3 and could not reproduce the error.

phargogh committed:
Specifying where tests are located.

phargogh committed:
Fixing tests for updated numpy warning message.

Numpy 1.23.0 updated their division-by-zero and
division-by-invalid-value warning messages, so this update handles them
both depending on the available numpy version.  RE:#266

phargogh commented:
@dcdenu4 I agree about the unfortunate nature of needing to manually expose things at that level, but I'm not sure that there's a way around it.  Even in the python stdlib, a bunch of the standard packages do the same thing so I guess this is normal.  As frustrating as it is, it feels like python is trying to give us granular control of what's exposed at the module level, which makes sense.

RE: Test failures, it turned out that numpy 1.23.0 (released recently) changed a warning message which meant the existing warning filters were invalid in one of the tests.  Both warning message formats are now handled.

dcdenu4 reviewed (approved):
Thanks for handling that @phargogh !
